contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -289,11 +289,7 @@ abstract contract PoSValidatorManager is
             revert UnauthorizedOwner(_msgSender());
         }
 
-        if (rewardRecipient == _msgSender()) {
-            delete $._rewardRecipients[validationID];
```

Comment:
I think delete's would still make sense. Looks like the gas cost of no-op delete is 100 (since delete just sets it to zero value) and gas cast of delete when a value was stored is 5000 but refunds up to 4900 (the gas cap being half of the total gas cast of the transaction)

Reply:
So I added the deletes back (this isn't actually the place where we wanted to do that btw), but I'm not quite sure I understand the point of deleting the value to begin with if there's a cost to deletion in both cases. If we were refunded more it would make sense, but once the validation is complete, is there any use for that field of the mapping? We won't use that `validationID` again, will we? 

Very possible I'm missing something here.

Reply:
Yeah sorry I just wanted to make a general point about deletes and put it in the wrong place. 

Also looks like I was wrong about the gas costs above as well as the refund cap. Refund cap is 20% of the overall tx gas cost and the refund is indeed higher than the cost in order to incentivize cleaning up after yourself. Playing around with https://www.evm.codes/#55 simulator it looks like in the actual delete case the cost would be: 2900 and refund 4800. The "warm" case should apply here since we check for the value first.

Reply:
Ha! To be honest, I figured that was the case which is why I added them back, but then I reread your comment which gave me pause. 

Definitely makes sense to delete when done here. Thanks for the comments!

Reply:
Well... Not "here", but you know what I mean